File: output/page_file_generator/old_file_v1/Equfix_Place_On_Alert.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

class Equfix_Place_On_Alert:
    LOCATORS = {
        "continue_button": (By.XPATH, "//button[@id='continue-button']"),
        "zip": (By.XPATH, "//input[@id='zip']"),
        "state_dropdown": (By.XPATH, "//button[@id='efx-dropdown-label-753393']"),
        "city": (By.XPATH, "//input[@id='city']"),
        "address_line2": (By.XPATH, "//input[@id='addressLine2Id']"),
        "address": (By.XPATH, "//input[@id='address']"),
        "phone_number": (By.XPATH, "//input[@id='phoneNumber']"),
        "ssn": (By.XPATH, "//input[@id='ssn']"),
        "dob": (By.XPATH, "//input[@id='dob']"),
        "last_name": (By.XPATH, "//input[@id='lastName']"),
        "first_name": (By.XPATH, "//input[@id='firstNameId']")
    }

    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 20)

    def wait_for_element(self, locator_key):
        locator = self.LOCATORS.get(locator_key)
        if locator is None:
            raise RuntimeError(f"Locator '{locator_key}' not found.")
        return self.wait.until(EC.presence_of_element_located(locator))

    def click(self, locator_key):
        element = self.wait_for_element(locator_key)
        element.click()

    def enter_text(self, locator_key, text):
        element = self.wait_for_element(locator_key)
        element.clear()
        element.send_keys(text)

    def switch_to_window_by_index(self, index):
        all_windows = self.driver.window_handles
        if index >= len(all_windows):
            raise RuntimeError(f"Invalid index: {index}. Total open windows: {len(all_windows)}.")
        self.driver.switch_to.window(all_windows[index])

    def switch_to_window_by_handle(self, handle):
        all_windows = self.driver.window_handles
        if handle not in all_windows:
            raise RuntimeError(f"Window handle '{handle}' not found.")
        self.driver.switch_to.window(handle)

    def switch_to_window_matching_url(self, url):
        all_windows = self.driver.window_handles
        for handle in all_windows:
            self.driver.switch_to.window(handle)
            if self.driver.current_url == url:
                return
        raise RuntimeError(f"No window found with URL '{url}'.")

    def switch_to_new_window(self, url):
        current_windows = set(self.driver.window_handles)
        self.wait.until(lambda driver: len(set(driver.window_handles) - current_windows) > 0)
        new_windows = set(self.driver.window_handles) - current_windows
        if not new_windows:
            raise RuntimeError("No new window appeared.")
        new_window_handle = new_windows.pop()
        self.driver.switch_to.window(new_window_handle)
        if self.driver.current_url != url:
            raise RuntimeError(f"Expected URL '{url}', but found '{self.driver.current_url}'.")

    def switch_to_personal_credit_fraud_page(self):
        self.switch_to_new_window("https://www.equifax.com/personal/credit-report-services/credit-fraud-alerts/")

    def fill_first_name(self, first_name):
        self.wait_for_element("first_name")
        self.enter_text("first_name", first_name)

    def fill_last_name(self, last_name):
        self.wait_for_element("last_name")
        self.click("last_name")
        self.enter_text("last_name", last_name)

    def fill_ssn(self, ssn_text):
        self.wait_for_element("ssn")

File: output/page_file_generator/old_file_v1/Equfix_Place_On_Alert.java
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.By;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;

import java.time.Duration;
import java.util.ArrayList;
import java.util.List;

public class Equfix_Place_On_Alert {

    private WebDriver driver;
    private WebDriverWait wait;

    @FindBy(how = How.XPATH, using = "//button[@id='continue-button']")
    private WebElement continueButton;

    @FindBy(how = How.XPATH, using = "//input[@id='zip']")
    private WebElement zip;

    @FindBy(how = How.XPATH, using = "//button[@id='efx-dropdown-label-753393']")
    private WebElement dropdownLabel;

    @FindBy(how = How.XPATH, using = "//input[@id='city']")
    private WebElement city;

    @FindBy(how = How.XPATH, using = "//input[@id='addressLine2Id']")
    private WebElement addressLine2;

    @FindBy(how = How.XPATH, using = "//input[@id='address']")
    private WebElement address;

    @FindBy(how = How.XPATH, using = "//input[@id='phoneNumber']")
    private WebElement phoneNumber;

    @FindBy(how = How.XPATH, using = "//input[@id='ssn']")
    private WebElement ssn;

    @FindBy(how = How.XPATH, using = "//input[@id='dob']")
    private WebElement dob;

    @FindBy(how = How.XPATH, using = "//input[@id='lastName']")
    private WebElement lastName;

    @FindBy(how = How.XPATH, using = "//input[@id='firstNameId']")
    private WebElement firstName;

    public Equfix_Place_On_Alert(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(20));
        PageFactory.initElements(driver, this);
    }

    private void waitForElement(WebElement element) {
        wait.until(ExpectedConditions.visibilityOf(element));
    }

    private void click(WebElement element) {
        waitForElement(element);
        element.click();
    }

    private void enterText(WebElement element, String text) {
        waitForElement(element);
        element.clear();
        element.sendKeys(text);
    }

    private void switchToWindowByIndex(int index) {
        List<String> windowHandles = new ArrayList<>(driver.getWindowHandles());
        if (index < windowHandles.size()) {
            driver.switchTo().window(windowHandles.get(index));
        }
    }

    private void switchToWindowByHandle(String handle) {
        driver.switchTo().window(handle);
    }

    private void switchToWindowMatchingUrl(String url) {
        for (String handle : driver.getWindowHandles()) {
            driver.switchTo().window(handle);
            if (driver.getCurrentUrl().equals(url)) {
                return;
            }
        }
    }

    private void switchToNewWindow(String url) {
        List<String> existingHandles = new ArrayList<>(driver.getWindowHandles());
        driver.getWindowHandles().stream().filter(handle -> !existingHandles.contains(handle)).findFirst().ifPresent(handle -> {
            driver.switchTo().window(handle);
            if (!driver.getCurrentUrl().equals(url)) {
                throw new RuntimeException("The new window URL does not match the expected URL: " + url);
            }
        });
    }

    public void clickContinueButton() {
        click(continueButton);
    }

    public void enterZipCode(String text) {
        enterText(zip, text);
    }

    public void clickDropdownLabel() {
        click(dropdownLabel);
    }

    public void enterCity(String text) {
        enterText(city, text);
    }

    public void enterAddressLine2(String text) {
        enterText(addressLine2, text);
    }

    public void enterAddress(String text) {
        enterText(address, text);
    }

    public void enterPhoneNumber(String text) {
        enterText(phoneNumber, text);
    }

    public void enterSSN(String text) {
        enterText(ssn, text);
    }

    public void enterDob(String text) {
        enterText(dob, text);
    }

    public void enterLastName(String text) {
        enterText(lastName, text);
    }

    public void enterFirstName(String text) {
        enterText(firstName, text);
    }

    public void performEquifixFlowActions() {
        // Example implementation for complex recorded actions:
        // This method will sequence all interactions from your recorded actions
        enterFirstName("test");
        click(ssn);
        enterSSN("***-**-6686");
        click(lastName);
        enterLastName("test");
        click(phoneNumber);
        enterPhoneNumber("786-876-****");
        click(dob);
        enterDob("04/22/1990");
        click(address);
        enterAddress("test");
        click(city);
        enterCity("test");
        click(addressLine2);
        enterAddressLine2("test");
        // Similarly, continue other actions from recorded actions
    }
}